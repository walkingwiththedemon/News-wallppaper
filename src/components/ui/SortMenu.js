import PropTypes from 'prop-types'

const options = {
    date: 'SORTED_BY_DATE',
    like: 'SORTED_BY_LIKE',
    comment: 'SORTED_BY_COMMENT'
}

const SortMenu = ({ sort='SORTED_BY_DATE', onSelect=f=>f }) =>
    <div className="page__sort">
        <div className="container">
            <div className="sort-menu">
                {
                    Object.keys(options).map((item, i) =>
                        <div key={i} className="sort-menu__item">
                            <button
                                className={(sort === options[item]) ? 'button button--state--active sort-menu__button' : 'button sort-menu__button'}
                                onClick={e => {
                                    e.preventDefault()
                                    onSelect(options[item])}}
                            >{item}</button>
                        </div>
                    )
                }
            </div>
        </div>
    </div>

SortMenu.propTypes = {
    sort: PropTypes.string,
    onSelect: PropTypes.func
}

export default SortMenu