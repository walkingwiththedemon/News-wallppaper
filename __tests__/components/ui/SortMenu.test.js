import toJSON from 'enzyme-to-json'
import { compose } from 'redux'
import SortMenu from '../../../src/components/ui/SortMenu'

const { shallow, mount } = Enzyme

describe("<SortMenu /> UI Component", () => {

    it("Renders Correctly", () =>
        compose(expect,toJSON,shallow)(
            <SortMenu sort="SORTED_BY_LIKE" />
        ).toMatchSnapshot()
    )

})