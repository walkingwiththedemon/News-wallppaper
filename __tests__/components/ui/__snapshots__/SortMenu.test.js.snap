// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortMenu /> UI Component Renders Correctly 1`] = `
<div
  className="page__sort"
>
  <div
    className="container"
  >
    <div
      className="sort-menu"
    >
      <div
        className="sort-menu__item"
        key="0"
      >
        <button
          className="button sort-menu__button"
          onClick={[Function]}
        >
          date
        </button>
      </div>
      <div
        className="sort-menu__item"
        key="1"
      >
        <button
          className="button button--state--active sort-menu__button"
          onClick={[Function]}
        >
          like
        </button>
      </div>
      <div
        className="sort-menu__item"
        key="2"
      >
        <button
          className="button sort-menu__button"
          onClick={[Function]}
        >
          comment
        </button>
      </div>
    </div>
  </div>
</div>
`;
